### PR TITLE
[7160] Change default data size to -1 for dstream (main)

### PR DESCRIPTION
**Pull request by Nishant2022**

### CODE REVIEW

**Comment by trel** [comment]
maybe a comment about our newly defined policy of setting dataSize to an out-of-range number b/c streaming...

File: lib/core/include/irods/transport/default_transport.hpp [hunk 1]
```diff
@@ -341,6 +341,7 @@ namespace irods::experimental::io::NAMESPACE_IMPL
             }
 
             dataObjInp_t input{};
+            input.dataSize = -1;
```
Comment (korydraughn) [inline lib/core/include/irods/transport/default_transport.hpp:344]: Let's move this down to the lines where we set `createMode` and `openFlags`.

File: lib/core/include/irods/transport/default_transport.hpp [hunk 2]
```diff
@@ -343,6 +343,8 @@ namespace irods::experimental::io::NAMESPACE_IMPL
             dataObjInp_t input{};
             at_scope_exit free_memory{[&input] { clearKeyVal(&input.condInput); }};
 
+            // Set dataSize to invalid size to signal unknown size for streaming
```
Comment (korydraughn) [inline lib/core/include/irods/transport/default_transport.hpp:346]: Need a period at the end of comment.

**Review by korydraughn — COMMENTED** [review]
Once comment is resolved, please squash w/o the pound.

**Comment by Nishant2022** [comment]
Squashed

**Comment by korydraughn** [comment]
Do the unit tests and core tests pass?

**Comment by Nishant2022** [comment]
All unit test pass.

All core tests passed apart from `test_control_plane` which is documented. Interestingly enough, `test_rulebase` passed, but then I ran `test_rulebase` again and it failed.

**Review by korydraughn — APPROVED** [review]
Add the pound when ready.

**Comment by Nishant2022** [comment]
Added pound

**Review by alanking — APPROVED** [review]

**Pull request merged**